### PR TITLE
Initial implementation.

### PR DESCRIPTION
**Pull request by sjrd**
Goes together with https://github.com/scala-js/scala-js/pull/4659

Draft because the CI script mentions my fork of scala-js/scala-js. We'll need it to change that once https://github.com/scala-js/scala-js/pull/4659 is merged.

But otherwise it's ready to review.

### CODE REVIEW

**Comment by sjrd** [comment]
Green CI build here: https://github.com/sjrd/scala-js-java-securerandom/actions/runs/2065025222

File: README.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
...
+
+```scala
+libraryDependencies += ("org.scala-js" %%% "scalajs-java-securerandom" % "1.0.0-SNAPSHOT").cross(CrossVersion.for3Use2_13)
+```
+
+When using a `crossProject`, add the above in `.jsSettings(...)`.
+
+You can then use `java.security.SecureRandom` from your code, and by extension, the `java.util.UUID.randomUUID()` method.
+
+When running on environments other than browsers or Node.js, attempts to use the above features will throw a `java.lang.UnsupportedOperationException`.
```
Comment (armanbilge) [inline README.md]: This is somewhat pessimistic. Other runtimes such as Cloudflare Workers (serverless) and Deno have also implemented the WebCrypto APIs.
https://developers.cloudflare.com/workers/runtime-apis/web-crypto/#methods
https://deno.com/blog/v1.11#more-web-crypto-apis-supported

Comment (sjrd) [reply README.md]: Is it better now?

Comment (armanbilge) [reply README.md]: Yes! Thanks :)

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  )
+
+lazy val testSuite = crossProject(JSPlatform, JVMPlatform)
+  .in(file("test-suite"))
+  .jsConfigure(_.enablePlugins(ScalaJSJUnitPlugin))
+  .enablePlugins(BuildInfoPlugin)
+  .settings(commonSettings: _*)
+  .settings(
+    testOptions += Tests.Argument(TestFramework("com.novocode.junit.JUnitFramework"), "-v", "-a"),
+    buildInfoKeys := Seq(BuildInfoKey("expectFailure" -> (jsEnvKind.value == JSEnvKind.JSDOMNodeJS))),
```
Comment (gzm0) [inline build.sbt]: Consider passing in the raw info so the logic if *why* we expect a failure remains encapsulated in the relevant test.

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    Compile / packageBin / mappings ~= {
+      _.filter(!_._2.endsWith(".class"))
+    },
+    exportJars := true,
+  )
+
+lazy val testSuite = crossProject(JSPlatform, JVMPlatform)
+  .in(file("test-suite"))
+  .jsConfigure(_.enablePlugins(ScalaJSJUnitPlugin))
+  .enablePlugins(BuildInfoPlugin)
```
Comment (gzm0) [inline build.sbt:133]: Side note: We should use this for the core repo as well. Right now, we roll our own :)

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/SecureRandomTest.scala [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+     */
+
+    val random = new SecureRandom()
+    for (_ <- 0 until 100) {
+      val f = random.nextFloat()
+      assertTrue(f >= 0.0f && f < 1.0f)
+    }
+  }
+
+  @Test def verifyFailure(): Unit = {
```
Comment (gzm0) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/SecureRandomTest.scala:67]: use `@Test(expect = classOf[UnsupportedOperationException)`? (also in UUIDTest)

**Review by gzm0 — CHANGES_REQUESTED** [review]
All comments from https://github.com/scala-js/scala-js-fake-insecure-java-securerandom/pull/1 also apply here.

File: project/plugins.sbt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+libraryDependencies += "org.scala-js" %% "scalajs-env-selenium" % "1.1.1"
+libraryDependencies += "org.scala-js" %% "scalajs-env-jsdom-nodejs" % "1.1.0"
+
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.10.0-SNAPSHOT")
+addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "1.2.0")
+
+addSbtPlugin("de.heikoseeberger" % "sbt-header" % "5.6.0")
+addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.10.0")
```
Comment (gzm0) [inline project/plugins.sbt]: Use 0.11.0?

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/SecureRandomTest.scala (hunk 4 above)
Comment (sjrd) [reply test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/security/SecureRandomTest.scala:67]: That doesn't appear to work in this case. When the assumption fails, it fails with
```
Unexpected exception, expected<java.lang.UnsupportedOperationException> but was<org.junit.AssumptionViolatedException>
```
So we have to keep the manual thing.

**Comment by sjrd** [comment]
I think I have addressed all the comments here as well, now.

**Comment by sjrd** [comment]
Upgraded to Scala.js 1.10.0 final.
Green CI build: https://github.com/sjrd/scala-js-java-securerandom/actions/runs/2090651909

**Review by gzm0 — APPROVED** [review]

**Pull request merged**